### PR TITLE
[Startrail-735][Startrail-728][Startrail-729] v1.3.0 subgraph changes

### PR DESCRIPTION
**Pull request by noyonthe1**
- [x] approval event added
- [x] provenance modified
- [x] bulk events modified

### CODE REVIEW

File: src/srr-mapping.ts
Comment (chatch) [inline src/srr-mapping.ts:94]: @noyonthe1 

you will get a duplicate exception here for a 2nd approval. 

actually in the schema you have a comment on ID `  id: ID!                      # keccak256(tokenId,timestamp)` and that wlil fix it actually. add the timestamp into the hash.

Comment (noyonthe1) [reply src/srr-mapping.ts:94]: oh ya, right, it's my mistake, I was about to add a timestamp then forgot after getting the error into the event. thanks for mentioning it. @chatch

Comment (noyonthe1) [reply src/srr-mapping.ts]: timestamp added into this commit https://github.com/startbahn/startrail-subgraph/pull/48/commits/333f2c6806a94fcf40a7ca316aded262aff9eaa7

**Comment by noyonthe1** [comment]
@chatch this PR is ready to review, please check.

**Review by chatch — CHANGES_REQUESTED** [review]

File: schema.graphql
Comment (chatch) [reply schema.graphql:121]: oh i see it's actually `issuer` when we store it in state but the name changes to `sender` on the event parameter name ..

File: src/srr-mapping.ts
Comment (noyonthe1) [reply src/srr-mapping.ts:187]: ya right, will fix it into the next commit.

File: schema.graphql
Comment (noyonthe1) [reply schema.graphql:122]: `bull's eye` thanks.

File: schema.graphql
Comment (noyonthe1) [reply schema.graphql:121]: okay sure, let me change that into next commit.

File: subgraph.template.yaml
Comment (noyonthe1) [reply subgraph.template.yaml:123]: @chatch It's been implemented into this commit, https://github.com/startbahn/startrail-subgraph/pull/48/commits/e4922af85a59871f4164f607e4dae77abf689cc1

File: subgraph.template.yaml
Comment (noyonthe1) [reply subgraph.template.yaml:125]: https://github.com/startbahn/startrail-subgraph/pull/48#discussion_r577277110

File: subgraph.template.yaml
Comment (noyonthe1) [reply subgraph.template.yaml:54]: https://github.com/startbahn/startrail-subgraph/pull/48#discussion_r577277110

File: schema.graphql
Comment (noyonthe1) [reply schema.graphql]: I have changed it into schema but just so that you know, the sender is defined into event params.

File: subgraph.template.yaml
Comment (chatch) [inline subgraph.template.yaml:48]: @noyonthe1 There is a lot of overlap between the legacy and non-legacy entries. What do you think about putting only legacy events in the legacy entry and the current entry has the full set of current events only? For each of the contracts the same pattern would apply.

In terms of ordering of contract entries in the template yaml. Would it make sense to put like contract entries together with legacy always coming second? That way as you scan down they are grouped together but recent entry comes first.

File: src/bulk-mapping.ts
Comment (chatch) [inline src/bulk-mapping.ts:23]: nice. yeah this is great.

File: subgraph.template.yaml
Comment (noyonthe1) [reply subgraph.template.yaml]: done into this commit https://github.com/startbahn/startrail-subgraph/pull/48/commits/734c27ed5c2496e90df6ec9c72632bb04e2ebefe

**Review by chatch — APPROVED** [review]
great thanks @noyonthe1 . approving after latest commits.

**Pull request merged**
